be/src/exec/pipeline/exchange/local_exchange_memory_manager.h

```diff
@@ -42,9 +44,11 @@ class LocalExchangeMemoryManager {
 
     bool is_full() const { return _memory_usage >= _max_memory_usage; }
 
+    void update_max_memory_usage(size_t limit) { _max_memory_usage = limit; }
```

Comment:
`_max_memory_usage` should be greater than `_max_memory_usage_per_driver*dop`. Otherwise, dead lock may occur, where the local exchange sink thinks the buffer is full, but the source thinks there isn't enough data.


```C++
bool LocalExchangeSourceOperator::has_output() const {
    std::lock_guard<std::mutex> l(_chunk_lock);

    return !_full_chunk_queue.empty() || _partition_rows_num >= _factory->runtime_state()->chunk_size() ||
           _key_partition_max_rows() >= _factory->runtime_state()->chunk_size() ||
           (_is_finished && (_partition_rows_num > 0 || _key_partition_max_rows() > 0)) || _local_buffer_almost_full();
}

bool _local_buffer_almost_full() const {
    return _local_memory_usage >= _memory_manager->get_memory_limit_per_driver() * 0.8;
}
```